### PR TITLE
Text selection refactoring

### PR DESCRIPTION
**Pull request by kirlat**
This build introduces a refactoring a text selection. Almost all business logic related to the selection has been moved to a new `SelectionController`. There were also some related changes to `UIEventController` and pointer events classes to support an extended functionality such as hot replacement of event parameters. I hope that those changes have allowed to simplify a business logic of the selection and made it more understandable.

This part is almost complete, but there are still some improvements to be made. For example, I think that it would make sens to move creation of an `HTMLSelector` from `Lexis` to a `SelectionController`. Then a latter will return an  `HTMLSelector` ready to be processed by a lexical query. 

The other part of this build is movement of all UI-related API to the UI controller. There are also some improvements to the UI logic.

I would like to present it for review so you can check a concept and let me know what do you think about it. If it's good, I will do other, much smaller, changes and will add tests. If not, I will make amends.

Also please see #482 for some additional information.

Thanks!

### CODE REVIEW

File: packages/components/src/lib/controllers/ui-controller.js [hunk 1]
```diff
@@ -62,10 +63,15 @@ export default class UIController {
...
+    // An object that stores config settings of a controller
+    this._config = {
+      overrideHelp
+    }
+
     // Options that are shown in a UI section of a Settings tab of the panel and control the visual representation
     this._uiOptions = null
 
-    this.TAB_NAMES_DEFAULT = defaultTabName
+    this.TAB_NAMES_DEFAULT = this._config.overrideHelp ? 'settings' : 'info'
```
[the remark this answers is not in the document]
Comment (kirlat) [reply packages/components/src/lib/controllers/ui-controller.js:74]: I think it is a good idea to add an explanation to that piece of code. It seems that in the "regular" workflow a default tab is `info`. But if  `overrideHelp` is on, then that would become a `settings` tab. I believe an idea is to let the client (embed lib, webextension) to override a default tab setting. 

@balmas, do you remember from where those requirements were originated? Maybe we can simplify this by letting the client specify directly what a default tab should be by passing a default tab name to the App Controller's options?

[the remark this answers is not in the document]
Comment (balmas) [reply packages/components/src/lib/controllers/ui-controller.js:74]: Ah, this is definitely a good candidate for cleanup.  The info tab effectively serves as the only help function in the UI right now.  However that text is specific to the features that we enable in the Alpheios webextension.  Users of the embedded library may not want to show this same help screen - in fact in our own usage of it in the alpheios reader, we use the overrideHelp setting to disable the info tab and have the info icon instead bring up the reader help popup.

File: packages/components/src/lib/controllers/ui-event-controller.js [hunk 2]
```diff
@@ -27,66 +32,119 @@ export default class UIEventController {
...
+    if (this._listeners.has(name)) {
       this.deactivateListener(name)
-      this.listeners.delete(name)
+      this._listeners.delete(name)
+    }
+    return this
+  }
+
+  /**
+   * Updates events for a listener.
```
[the remark this answers is not in the document]
Comment (kirlat) [reply packages/components/src/lib/controllers/ui-event-controller.js:58]: Yes, you've understood it correctly. I will update a description.

File: packages/components/src/lib/custom-pointer-events/mouse-move.js [hunk 3]
```diff
@@ -4,16 +4,28 @@ import Platform from '@/lib/utility/platform.js'
...
     this.boundListener = this.eventListener.bind(this) // A bound event listener of MouseDblClick
-    this.mouseMoveDelay = parseInt(eventParams.mouseMoveDelay)
-    this.mouseMoveAccuracy = parseInt(eventParams.mouseMoveAccuracy)
+    this.mouseMoveDelay = parseInt(mouseMoveDelay)
+    this.mouseMoveAccuracy = parseInt(mouseMoveAccuracy)
+    this.enableMouseMoveLimitedByIdCheck = enableMouseMoveLimitedByIdCheck
+    this.mouseMoveLimitedById = mouseMoveLimitedById
+  }
+
+  // TODO: It is not used right now. Do we need to keep it?
```
[the remark this answers is not in the document]
Comment (kirlat) [reply packages/components/src/lib/custom-pointer-events/mouse-move.js:19]: I think I was wrong here. It is used in a parent class, `PointerEvt`. That got me confused. I will remove my comment then.

**Pull request merged**